src/redux/books/books.js

```diff
@@ -0,0 +1,32 @@
+const actions = {
+  ADD_BOOK: 'ADD_BOOK',
+  DELETE_BOOK: 'DELETE_BOOK',
+};
+
+export default function reducer(state = [], action) {
+  switch (action.type) {
+    case actions.ADD_BOOK:
+      return [
+        ...state,
+        {
+          title: action.title,
+          author: action.author,
+        },
+      ];
+    case actions.DELETE_BOOK:
+      return state.map((book) => book.id !== action.id);
```

Comment:
- [ ] I don't think this expression does what you intend. The `map()` method here merely returns `true` or `false`, whereas we need to return the book objects themselves - and the `filter()` method does just that. Kindly use the filter method here instead.

  ![map-filter](https://user-images.githubusercontent.com/36542986/178257257-393964f0-5a1a-4217-b1b9-a8922dd6737c.jpg)